### PR TITLE
EPMRPP-46991. Unit tests.

### PR DESCRIPTION
**Pull request by ElenaRomanchuk**

### CODE REVIEW

File: test/reporter.test.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+  afterEach(() => {
+    reporter.testItemIds.clear();
+    global.Date = RealDate;
+    jest.clearAllMocks();
+  });
+  it('constructor: client should be defined', () => {
+    expect(reporter.client).toBeDefined();
+  });
+
+  it('runStart: startLaunch should be called with parameters', () => {
```
Comment (KatsiarynaTatarynovich) [inline test/reporter.test.js]: What do you think of splitting different functions into different nested describes?
```
reporter script
|-runStart
  |-test
  |-test
  |-...
|-runEnd
  |-...
|-....
  |-...
```

File: test/reporter.test.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    reporter.runStart(launchObj);
+
+    expect(reporter.tempLaunchId).toEqual('tempLaunchId');
+    expect(spyLaunchStart).toHaveBeenCalledTimes(1);
+    expect(spyLaunchStart).toHaveBeenCalledWith(launchObj);
+  });
+
+  it('runEnd: finishLaunch should be called with parameters', () => {
+    const spyFinishLaunch = jest.spyOn(reporter.client, 'finishLaunch');
+    reporter.tempLaunchId = 'tempLaunchId';
```
Comment (KatsiarynaTatarynovich) [inline test/reporter.test.js]: I think it's a good practice to add separation between your test structure. 
Like in the example below 
https://github.com/goldbergyoni/javascript-testing-best-practices#-%EF%B8%8F-12-structure-tests-by-the-aaa-pattern

File: test/reporter.test.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+      parentId: undefined,
+    };
+    reporter.suiteStart(suiteStartObject);
+    expect(spyStartTestItem).toHaveBeenCalledTimes(1);
+    expect(spyStartTestItem).toHaveBeenCalledWith(suiteStartObject, 'tempLaunchId', undefined);
+  });
+
+  it('suiteStart for nested suite: startTestItem should be called with parameters', () => {
+    const spyStartTestItem = jest.spyOn(reporter.client, 'startTestItem');
+    reporter.tempLaunchId = 'tempLaunchId';
```
Comment (KatsiarynaTatarynovich) [inline test/reporter.test.js]: You should reset such value and the other global value's after tests in the corresponding "after" methods

Comment (AmsterGet) [reply test/reporter.test.js]: Should this be done for all tests in this suite? I don't think so )

**Pull request merged**